docs/custom_instrumentation.md

```diff
@@ -0,0 +1,171 @@
+# Custom Instrumentation
+
+This guide shows you how to add custom telemetry to your iOS app using the AWS Distro for OpenTelemetry Swift.
+
+## Custom Spans
+
+Spans track operations in your app. Use them to measure how long something takes.
+
+### Basic Span
+
+```swift
+import OpenTelemetryApi
+
+let tracer = OpenTelemetry.instance.tracerProvider.get(instrumentationName: "my-app")
+
+let span = tracer.spanBuilder(spanName: "load_user_profile").startSpan()
+defer { span.end() }
+
+let user = database.getUser(userId: userId)
+processUser(user)
+```
+
+### Span with Attributes
+
+Add context to your spans with attributes:
+
+```swift
+func checkoutButton() {
+    let span = tracer.spanBuilder(spanName: "checkout")
+        .setAttribute(key: "cart.items", value: 3)
+        .setAttribute(key: "cart.total", value: 49.99)
+        .setAttribute(key: "payment.method", value: "credit_card")
+        .startSpan()
+    defer { span.end() } // close the span on function end
+
+    processCheckout()
+}
+```
+
+### Async Span Control
+
+For async operations:
+
+```swift
+func performBackgroundTask() async {
+    let span = tracer.spanBuilder(spanName: "background_task").startSpan()
+
+    do {
+        try await doWork()
+    } catch {
+        span.recordException(error)
+    }
+
+    span.end()
+}
+```
+
+## Custom Events
+
+Events are point-in-time occurrences. Use them to track user actions or important moments.
+
+### Basic Event
+
+```swift
+import OpenTelemetryApi
+
+let logger = OpenTelemetry.instance.loggerProvider.get(instrumentationScopeName: "my-app")
+
+logger.eventBuilder(name: "button_clicked")
```

Comment:
issue: eventBuilder was deprecated in https://github.com/open-telemetry/opentelemetry-swift-core/pull/15